app/controllers/name_controller/create_and_edit_name.rb

```diff
@@ -112,13 +104,31 @@ def set_locked_if_admin
   end
 
   def set_icn_id_if_unlocked_or_admin
-    @name.icn_id   = params[:name][:icn_id] if name_unlocked? || in_admin_mode?
+    @name.icn_id   = params[:name][:icn_id] if name_unlocked?
```

Comment:
Comment: I should probably rename `name_unlocked?`, which returns true if `in_admin_mode?`

Reply:
Oh gosh, yes, that does feel dangerous.  I wouldn't be surprised if I were responsible for that.  You're thinking "name_unlocked_or_in_admin_mode?"... wait, isn't there already a "can_edit?" or something similar?  That may already be checking if the name is locked and admin mode.

Reply:
There *is* a Name#can_edit? but it is just inherited from AbstractModel and it does not check Name#locked like it should.  Unfortunately, I don't think non-controllers have access to the session and therefore cannot know whether admin mode is turned on.

Reply:
How about `editable?`?

Reply:
Maybe in AbstractController
```ruby
  def editable_in_session?(obj)
    in_admin_mode? || obj.can_edit?(@user)  
  end
```  
Then call it with `editable_in_session?(@name)`

Reply:
No. that's not right because it can flash an error.
For now I'll keep it simple by changing the name of `CreateAndEdit#name_unlocked` to `editable_in_session`.